fastly/block_fastly_service_v1_s3logging.go

```diff
@@ -0,0 +1,331 @@
+package fastly
+
+import (
+	"fmt"
+	"log"
+	"strings"
+
+	gofastly "github.com/fastly/go-fastly/fastly"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+)
+
+var s3loggingSchema = &schema.Schema{
+	Type:     schema.TypeSet,
+	Optional: true,
+	Elem: &schema.Resource{
+		Schema: map[string]*schema.Schema{
+			// Required fields
+			"name": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "The unique name of the S3 logging endpoint.",
+			},
+
+			"bucket_name": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "S3 Bucket name to store logs in.",
+			},
+
+			"s3_access_key": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				DefaultFunc: schema.EnvDefaultFunc("FASTLY_S3_ACCESS_KEY", ""),
+				Description: "AWS Access Key.",
+				Sensitive:   true,
+			},
+
+			"s3_secret_key": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				DefaultFunc: schema.EnvDefaultFunc("FASTLY_S3_SECRET_KEY", ""),
+				Description: "AWS Secret Key.",
+				Sensitive:   true,
+			},
+
+			// Optional fields
+			"path": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "Path to store the files. Must end with a trailing slash.",
```

Comment:
Wondering if we should have a validation function for this. I guess it would depend on how the API handles this --- i.e., does it validate this?

Reply:
The backend does validate and ensure that a trailing slash is appended if it doesn't exist.

Reply:
So the API appends a trailing slash if one is present? This would result in a diff always showing in the TF plan. As the statefile will have without and the API will have with?

Reply:
The API does append a trailing slash, but it wouldn't result in a diff in the TF plan because the infra state that gets stored is what is returned from the API.

In other words, a CREATE also calls a READ, which is what sets the TF state.